src/router/index.ts

```diff
@@ -36,4 +54,19 @@ const router = new VueRouter({
   routes
 })
 
+router.beforeEach((to, from, next) => {
+  const middleware = to.meta.middleware
+
+  if (!middleware) {
+    return next()
+  }
+
+  return middleware[0]({
+    to,
+    from,
+    next,
+    store
```

Comment:
Can't store be imported inside of middleware files? That way you will choose which middleware uses the store and which doesn't